idaes/tests/test_import.py

```diff
@@ -11,81 +11,285 @@
+    'parametrize' decorator.
     """
-    base, failures, total = root.parent, {}, 0
-    # iterate over flattened list of all paths ending a Python source file
-    for path in root.rglob("*.py"):
-        # check that all path components are valid module names
-        # - this is to skip directories like '.ipynb_checkpoints'
-        bad = False
-        for name in path.parts[:-1]:
-            if name == os.path.sep:
-                continue
-            if (
-                name.startswith("Workshop")       # skip workshop tutorials (1)
-                or name.startswith("Module_")     # skip workshop tutorials (2)
-                or name == "tests"                # skip all tests
-                or not good_modname.match(name)   # no _bla
-            ):
-                bad = True
-
-            if bad:
-                break
-        # stop if bad directory component or Python filename is invalid
-        bad = bad or not good_modname.match(path.parts[-1][:-3])
-        if bad:
-            print_path_status(path, SKIP)
-            continue
-        # module is valid, try importing it now
-        module_path = path.relative_to(base).with_suffix("")
-        module_name = ".".join(module_path.parts)
-        try:
-            start = time.time()
-            importlib.import_module(module_name)
-            sec = time.time() - start
-            if sec > max_sec:
-                print_path_status(path, BAD, msg="time={sec:3.1f}s")
-                raise ImportError(f"Import took too long ({sec:.1f}s)")
-            print_path_status(path, OK)
-        except ImportError as e:
-            failures[module_name] = str(e)
-            print_path_status(path, BAD, msg="import error")
-        total += 1
-    return failures, total
+    all_modules = sorted(
+        [
+            str(p.relative_to(idaes_path)).replace(os.sep, ".")[:-3]
+            for p in idaes_path.rglob("*.py")
+            if not p.name == "__init__.py"
+        ]
+    )
+    batched, num_modules = [], len(all_modules)
+    for i in range(0, num_modules, size):
+        batched.append(all_modules[i : min(i + size, num_modules)])
+    return batched
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("module_names", modules_batched())
+def test_import(module_names):
+    """Run the Python interpreter to import batches of modules.
+
+    Since most of the overhead of running this is in starting the interpreter,
+    it is much faster to do, e.g., 10 imports at a time -- as long as failure is
+    rare. The drawback is a little extra complexity if there is an error in an
+    import, since one needs to go back through the batch and see which module
+    actually caused the failure. Note that if failure is *not* rare, we have bigger
+    problems than whether this test is slow.
+    """
+    # Try to import everything in the batch, hope for success
+    batch_max = IMPORT_TIME_MAX * len(module_names) + 1
+    t0 = time.time()
```

Comment:
`time()` is not the most accurate timer.  I would recommend `time.perf_counter()`

Reply:
I'm sure you're right, but does a few milliseconds matter here?

Reply:
No - probably not (at least not when the threshold is 5 seconds - although that should probably be significantly reduced, but that is a different matter).

That said, timing in Python is noisy enough as it is, and `time()` can behave significant;y differently on different platforms.  In my experience, `perf_counter()` seems to have more stable behavior.